src/cvm/core/z3_types.cc

```diff
@@ -156,15 +156,40 @@ size_t Shape::Size() const {
 
 std::string Shape::to_string() const {
   std::ostringstream oss;
-  oss << "<";
+  oss << "(";
   for (auto it = begin(); it != end(); ++it) {
     if (it != begin()) oss << ", ";
     oss << *it;
   }
-  oss << ">";
+  oss << ")";
   return oss.str();
 }
 
+Shape Shape::from_string(const std::string& st) {
+  Shape re;
+  int stlen = st.length();
+  if (st.length() == 0){
```

Comment:
Left whitespace before braces.
```suggestion
  if (st.length() == 0) {
```